src/D2L.Bmx/WriteHandler.cs

```diff
@@ -64,39 +69,50 @@ bool useCredentialProcess
+		var awsConfig = GetParsedIniFileOrNull( awsConfigFilePath );
+
+		string awsCredentialsFilePath = useCredentialProcess ? SharedCredentialsFile.DefaultFilePath : output;
+		var awsCredentials = GetParsedIniFileOrNull( awsCredentialsFilePath );
+
 		if( useCredentialProcess ) {
-			string sectionName = $"profile {profile}";
-			if( !data.Sections.ContainsSection( sectionName ) ) {
-				data.Sections.AddSection( sectionName );
+			Debug.Assert( awsConfig is not null );
+
+			if( awsCredentials is not null && awsCredentials.Sections.ContainsSection( profile ) ) {
+				awsCredentials.Sections.RemoveSection( profile );
+				parser.WriteFile( awsCredentialsFilePath, awsCredentials, Utf8 );
 			}
-			if( File.Exists( SharedCredentialsFile.DefaultFilePath ) ) {
-				var defaultCredentialsFile = parser.ReadFile( SharedCredentialsFile.DefaultFilePath );
-				if( defaultCredentialsFile.Sections.ContainsSection( profile ) ) {
-					defaultCredentialsFile.Sections.RemoveSection( profile );
-					parser.WriteFile( SharedCredentialsFile.DefaultFilePath, defaultCredentialsFile );
-				}
+
+			string sectionName = profile == "default" ? "default" : $"profile {profile}";
+
+			if( !awsConfig.Sections.ContainsSection( sectionName ) ) {
+				awsConfig.Sections.AddSection( sectionName );
 			}
-			data[sectionName]["credential_process"] =
+			awsConfig[sectionName]["credential_process"] =
 				"bmx print --format json --cache --non-interactive"
 				+ $" --org {oktaApi.Org}"
 				+ $" --user {oktaApi.User}"
 				+ $" --account {awsCredsInfo.Account}"
 				+ $" --role {awsCredsInfo.Role}"
 				+ $" --duration {awsCredsInfo.Duration}";
+
+			parser.WriteFile( awsConfigFilePath, awsConfig, Utf8 );
 		} else {
-			if( File.Exists( SharedCredentialsFile.DefaultConfigFilePath ) ) {
-				string sectionName = $"profile {profile}";
-				var defaultConfigFile = parser.ReadFile( SharedCredentialsFile.DefaultConfigFilePath );
-				if( defaultConfigFile.Sections.ContainsSection( sectionName )
-					&& defaultConfigFile[sectionName].ContainsKey( "credential_process" ) ) {
-
-					if( defaultConfigFile[sectionName].Count == 1 ) {
-						defaultConfigFile.Sections.RemoveSection( sectionName );
+			Debug.Assert( awsCredentials is not null );
+
+			if( awsConfig is not null ) {
+				string sectionName = profile == "default" ? "default" : $"profile {profile}";
+
+				if(
+					awsConfig.Sections.ContainsSection( sectionName )
+					&& awsConfig[sectionName].ContainsKey( "credential_process" )
+				) {
+					if( awsConfig[sectionName].Count == 1 ) {
+						awsConfig.Sections.RemoveSection( sectionName );
 					} else {
-						defaultConfigFile[sectionName].RemoveKey( "credential_process" );
+						awsConfig[sectionName].RemoveKey( "credential_process" );
 					}
-					parser.WriteFile( SharedCredentialsFile.DefaultConfigFilePath, defaultConfigFile );
+					parser.WriteFile( awsConfigFilePath, awsConfig, Utf8 );
```

Comment:
nit: without explicitly specifying the encoding, `parser` will write file using UTF8-BOM 🤢